lib/core/text/flat.nit

```diff
@@ -1184,10 +1183,11 @@ redef class NativeString
 		return new FlatString.with_infos(self, len, 0)
 	end
 
-	# Returns `self` as a new String.
-	redef fun to_s_with_copy: FlatString
+	redef fun to_s_with_copy do return to_s_with_copy_and_length(cstring_length)
+
+	# Get a `String` from `length` bytes at `self` copied into Nit memory
+	fun to_s_with_copy_and_length(length: Int): String
```

Comment:
I propose to use some fine 2KY OO-design approach

~~~nit
class ToStringer
   fun copy(from: NativeString): String do ...
end
class ToStringerWithLength
  super ToStringer
  var length: Int
  redef fun copy(from) do ...
end
class ToStringerWithCopy
  super ToStringer
  redef fun copy(from) do ...
end
class ToStringerWithCopyAndLength
  super ToStingerWithLength
  super ToStingerWithCopy
end
~~~